### PR TITLE
utils restructured in subfolders and removed unused imports

### PR DESCRIPTION
**Pull request by allaffa**

### CODE REVIEW

**Comment by allaffa** [comment]
@pzhanggit @jychoi-hpc 
I think I restructured all the imports. 
Since this is a big PR, please let us take time to review it. 
I do not want to merge it before September 30. 
However, I would like to make gradual changes based on your feedback during this time.

**Comment by allaffa** [comment]
@RylieWeaver @AdithyaRaman @erdemcaliskan 

Please let me know if you need help on learning how to review a PR on GitHub.

File: hydragnn/utils/optimizer/optimizer.py
Comment (erdemcaliskan) [inline hydragnn/utils/optimizer/optimizer.py:1]: Not related to PR but in line 73 Adamax is equal to torch.optim.Adagrad, it should be equal to torch.optim.Adamax

File: hydragnn/utils/print/print_utils.py
Comment (erdemcaliskan) [inline hydragnn/utils/print/print_utils.py:1]: Just a small inconsistency - this file is print/print_utils.py whereas for some files we don't have utils in the file name e.g., model/model.py, optimizer/optimizer.py.

File: hydragnn/preprocess/stratified_sampling.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        dataset_categories.append(category)
+
+    subsample_indices = []
+    subsample = []
+
+    sss = StratifiedShuffleSplit(
+        n_splits=1, train_size=subsample_percentage, random_state=0
+    )
+
+    for subsample_index, rest_of_data_index in sss.split(dataset, dataset_categories):
```
Comment (erdemcaliskan) [inline hydragnn/preprocess/stratified_sampling.py:42]: _ may be more appropriate since rest_of_data_index is not used

**Review by erdemcaliskan — COMMENTED** [review]
Finally, we have a utils folder in the main and one more in hydragnn/utils which is confusing.

File: examples/multidataset/train.py [hunk 2]
```diff
@@ -130,8 +130,8 @@ def info(*args, logtype="info", sep=" "):
     )
 
     log_name = "GFM" if args.log is None else args.log
```
Comment (AdithyaRaman) [inline examples/multidataset/train.py:132]: Suggestion to append the run timestamp to log_name for easier differentiation across different runs:

```
log_name = "GFM" if args.log is None else args.log
current_time = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
log_name = log_name + "_" + current_time
```

File: hydragnn/utils/optimizer/optimizer.py
Comment (allaffa) [reply hydragnn/utils/optimizer/optimizer.py:1]: @erdemcaliskan 
Thank you. Even if it is not related to this PR, I fixed it anyway. :)

File: examples/qm7x/train.py [hunk 3]
```diff
@@ -58,15 +61,15 @@ def info(*args, logtype="info", sep=" "):
     getattr(logging, logtype)(sep.join(map(str, args)))
 
 
-from hydragnn.utils.abstractbasedataset import AbstractBaseDataset
+from hydragnn.utils.datasets import AbstractBaseDataset
```
[the remark this answers is not in the document]
Comment (allaffa) [reply examples/qm7x/train.py:64]: I disagree. When I replace the line with that you suggest, PyCharm says `no module named AbstractBaseDataset`

File: hydragnn/preprocess/stratified_sampling.py (hunk 1 above)
Comment (allaffa) [reply hydragnn/preprocess/stratified_sampling.py]: Thanks.

File: examples/qm7x/train.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (allaffa) [reply examples/qm7x/train.py]: Thanks.

File: examples/ogb/train_gap.py
[the remark this answers is not in the document]
Comment (allaffa) [reply examples/ogb/train_gap.py:1]: Thanks.

File: hydragnn/utils/datasets/__init__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from .abstractbasedataset import AbstractBaseDataset
```
[the remark this answers is not in the document]
Comment (allaffa) [reply hydragnn/utils/datasets/__init__.py:1]: This seems a nice restructuring, but as you said, it also seem to require a substantial restructuring which may deserve a standalone PR to be implemented.

**Comment by allaffa** [comment]
> Finally, we have a utils folder in the main and one more in hydragnn/utils which is confusing.

@erdemcaliskan 
I moved the `lsms` utils into the `hydragnn.utils` folder

File: hydragnn/utils/print/print_utils.py
Comment (erdemcaliskan) [reply hydragnn/utils/print/print_utils.py:1]: I see, makes sense that it wouldn't accept. Now that I checked we also have time_utils and smiles_utils. They can stay as they are or maybe _printing_ and _timing_ can be used.

Comment (allaffa) [reply hydragnn/utils/print/print_utils.py:1]: Let us keep the renaming as is. 
We spent already enough time on renaming discussions. 
I think we should move forward to other things.

**Comment by allaffa** [comment]
@pzhanggit 
I just solved very minor conflicts due to PRs recently merged for the GB effort. 
Please remember to send me your feedback by September 30.

**Comment by allaffa** [comment]
@RylieWeaver @pzhanggit @jychoi-hpc 
I rebased this PR after PR#278 has been merged into the main branch.

**Comment by RylieWeaver** [comment]
> @RylieWeaver @pzhanggit @jychoi-hpc I rebased this PR after PR#278 has been merged into the main branch.

Good, and I see you restructured the imports in the LennardJones example as well. It's hard to prove a negative, but this PR looks good to me in the parts of the codebase that I am familiar with.

**Review by RylieWeaver — APPROVED** [review]

**Comment by allaffa** [comment]
@pzhanggit @jychoi-hpc 

As planned more than a month ago, the PR will be merged at the end of the upcoming business week unless there are reviews that require additional changes.

File: examples/ising_model/train_ising.py
[the remark this answers is not in the document]
Comment (AdithyaRaman) [reply examples/ising_model/train_ising.py:1]: @allaffa @jychoi-hpc 
The adios2 imports inside other examples too (multidataset, multidataset_hpo, alexandria, lsms)  need to be updated to :
`hydragnn.utils.datasets.adiosdataset import AdiosWriter, AdiosDataset`

as pointed out by @erdemcaliskan. I believe this line update has been made only for the ising example

**Pull request merged**